### PR TITLE
Pool resource and data source with projects set

### PR DESCRIPTION
**Pull request by by-nelson**
### Changes
- Added set attribute to pool schema for both resource and data source.
- Added more test steps to verify project assignments.
- Updated pool example.

### Tests

Test case can be run with:

```sh
TF_ACC=1 go test -count=1 -run='TestAccPoolResource_basic' -v ./...   
```

```sh
?       terraform-provider-teamcity     [no test files]
?       terraform-provider-teamcity/models      [no test files]
testing: warning: no tests to run
PASS
ok      terraform-provider-teamcity/client      0.002s [no tests to run]
=== RUN   TestAccPoolResource_basic
--- PASS: TestAccPoolResource_basic (4.40s)
PASS
ok      terraform-provider-teamcity/teamcity    4.402s

```

### CODE REVIEW

File: teamcity/pool_data_source.go [hunk 1]
```diff
@@ -96,15 +102,29 @@ func (d *poolDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
...
 		state = models.PoolDataModel{
-			Name: types.StringValue(string(pool.Name)),
-			Size: types.Int64Value(int64(*(pool.Size))),
-			Id:   types.Int64Value(int64(*(pool.Id))),
+			Name:     types.StringValue(string(pool.Name)),
+			Size:     types.Int64Value(int64(*(pool.Size))),
+			Id:       types.Int64Value(int64(*(pool.Id))),
+			Projects: types.SetNull(types.StringType),
+		}
+	}
```
Comment (ChildOfJustice) [inline teamcity/pool_data_source.go:117]: [minor] I think if we only set size to a variable in if here we could remove some duplication

Something like
```
var size = basetypes.NewInt64Null()
    if pool.Size != nil {
        size = types.Int64Value(int64(*(pool.Size)))
	}
	
    state = models.PoolDataModel{
        Name:     types.StringValue(string(pool.Name)),
        Size:     size,
        Id:       types.Int64Value(int64(*(pool.Id))),
        Projects: types.SetNull(types.StringType),
    }

File: teamcity/pool_resource.go [hunk 2]
```diff
@@ -60,6 +63,17 @@ func (r *poolResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
 					int64validator.AtLeast(0),
 				},
 			},
+			"projects": schema.SetAttribute{
+				Computed:            true,
+				Required:            false,
+				Optional:            true,
+				MarkdownDescription: "Projects assigned to the given pool",
```
[the remark this answers is not in the document]
Comment (ChildOfJustice) [reply teamcity/pool_resource.go:70]:
```suggestion
				MarkdownDescription: "Projects assigned to the given pool. List of Project IDs.",
```

**Review by ChildOfJustice — CHANGES_REQUESTED** [review]
Thank you! Could you please take a look at my suggestions?

File: teamcity/pool_resource.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (by-nelson) [reply teamcity/pool_resource.go:70]: Yes you're right, I will commit your suggestion

File: teamcity/pool_resource.go [hunk 3]
```diff
@@ -105,6 +120,38 @@ func (r *poolResource) Create(ctx context.Context, req resource.CreateRequest, r
...
+	} else {
+		projects := []attr.Value{}
+		for _, p := range response.Project {
+			projects = append(projects, types.StringValue(string(*p.Id)))
+		}
+
+		plan.Projects, diags = types.SetValue(types.StringType, projects)
+		if diags.HasError() {
+			return
+		}
```
[the remark this answers is not in the document]
Comment (by-nelson) [reply teamcity/pool_resource.go]: I simplified it a little as you suggested, thanks!

File: teamcity/pool_data_source.go (hunk 1 above)
Comment (by-nelson) [reply teamcity/pool_data_source.go]: You're right, done!

**Review by ChildOfJustice — APPROVED** [review]
Thank you!

**Pull request merged**